Sources/FoundationEssentials/Subprocess/Platforms/Subprocess+Unix.swift

```diff
@@ -0,0 +1,286 @@
+            }
+
+            for path in searchPaths {
+                let fullPath = "\(path)/\(executableName)"
+                let fileExists = Subprocess.Configuration.pathAccessible(fullPath, mode: X_OK)
+                if fileExists {
+                    return fullPath
+                }
+            }
+        case .path(let executablePath):
+            // Use path directly
+            return executablePath.string
+        }
+        return nil
+    }
+}
+
+// MARK: - Configuration
+extension Subprocess.Configuration {
+    internal func preSpawn() throws -> (
+        executablePath: String,
+        env: [UnsafeMutablePointer<CChar>?],
+        argv: [UnsafeMutablePointer<CChar>?],
+        intendedWorkingDir: FilePath,
+        uidPtr: UnsafeMutablePointer<uid_t>?,
+        gidPtr: UnsafeMutablePointer<gid_t>?,
+        supplementaryGroups: [gid_t]?
+    ) {
+        // Prepare environment
+        let env = self.environment.createEnv()
+        // Prepare executable path
+        guard let executablePath = self.executable.resolveExecutablePath(
+            withPathValue: self.environment.pathValue()) else {
+            for ptr in env { ptr?.deallocate() }
+            throw CocoaError(.executableNotLoadable, userInfo: [
+                .debugDescriptionErrorKey : "\(self.executable.description) is not an executable"
+            ])
+        }
+        // Prepare arguments
+        let argv: [UnsafeMutablePointer<CChar>?] = self.arguments.createArgs(withExecutablePath: executablePath)
+        // Prepare workingDir
+        let intendedWorkingDir = self.workingDirectory
+        guard Self.pathAccessible(intendedWorkingDir.string, mode: F_OK) else {
+            for ptr in env { ptr?.deallocate() }
+            for ptr in argv { ptr?.deallocate() }
+            throw CocoaError(.fileNoSuchFile, userInfo: [
+                .debugDescriptionErrorKey : "Failed to set working directory to \(intendedWorkingDir)"
+            ])
+        }
+
+        var uidPtr: UnsafeMutablePointer<uid_t>? = nil
+        if let userID = self.platformOptions.userID {
+            uidPtr = .allocate(capacity: 1)
+            uidPtr?.pointee = uid_t(userID)
+        }
+        var gidPtr: UnsafeMutablePointer<gid_t>? = nil
+        if let groupID = self.platformOptions.groupID {
+            gidPtr = .allocate(capacity: 1)
+            gidPtr?.pointee = gid_t(groupID)
+        }
+        var supplementaryGroups: [gid_t]?
+        if let groupsValue = self.platformOptions.supplementaryGroups {
+            supplementaryGroups = groupsValue.map { gid_t($0) }
+        }
+        return (
+            executablePath: executablePath,
+            env: env, argv: argv,
+            intendedWorkingDir: intendedWorkingDir,
+            uidPtr: uidPtr, gidPtr: gidPtr,
+            supplementaryGroups: supplementaryGroups
+        )
+    }
+
+    internal static func pathAccessible(_ path: String, mode: Int32) -> Bool {
+        return path.withCString {
```

Comment:
Should we use `FileManager` to perform this here which will correctly use the `fileSystemRepresentation` instead of `withCString`?